### PR TITLE
Dashboard dropdown max visible items

### LINKED ISSUE #2566: Dashboard resource dropdowns max height
Resource dropdowns on various pages needs a maximum height. Currently the list of resources disappears off the screen:

image

### PR DESCRIPTION
**Pull request by MatsM16**
- Fixes #2566 
- Added new component `ResourceSelect`
- Resource and application dropdowns now have a scrollbar when more than `10` items.

## 9 or less items in the dropdown:
The popup scales to fit the items.
4 items in the dropdown

## 10 items in the dropdown:
The popup has scaled to its maximum size, but will not show a scrollbar
10 items in the dropdown

## 11 or more items in the dropdown:
The popup has scaled to its maximum size and shows a scrollbar.
11 items in the dropdown

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2678)

### CODE REVIEW

**Comment by MatsM16** [comment]
Rebased to latest `main`-branch.
Added `resource-list` class to `ResourceSelect` component to reflect changes to `main`.

File: src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public SelectViewModel<ResourceTypeDetails> SelectedResource { get; set; } = default!;
+
+    [Parameter]
+    public EventCallback<SelectViewModel<ResourceTypeDetails>> SelectedResourceChanged { get; set; }
+
+    [Parameter]
+    public string? AriaLabel { get; set; }
+
+    [Inject]
+    private IJSRuntime Js { get; set; } = default!;
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs:30]:
```suggestion
    public required IJSRuntime JSRuntime { get; init; }
```

File: src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    /// selected item doesn't update automatically when the item changes.
+    /// </summary>
+    public ValueTask UpdateDisplayValueAsync()
+    {
+        if (Js is null || _resourceSelectComponent is null)
+        {
+            return ValueTask.CompletedTask;
+        }
+
+        return Js.InvokeVoidAsync("updateSelectDisplayValue", _resourceSelectComponent.Element);
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs:45]:
```suggestion
        if (JSRuntime is null || _resourceSelectComponent is null)
        {
            return ValueTask.CompletedTask;
        }

        return JSRuntime.InvokeVoidAsync("updateSelectDisplayValue", _resourceSelectComponent.Element);
```

**Review by JamesNK — APPROVED** [review]
Looks good. Recommended a minor change to be consistent with the rest of the app

File: src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace Aspire.Dashboard.Components.Controls;
+
+public partial class ResourceSelect
+{
+    private const int ResourceOptionPixelHeight = 32;
+    private const int MaxVisibleResourceOptions = 10;
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs:15]: I played around with it and 10 felt too small. Trying a larger value.
```suggestion
    private const int MaxVisibleResourceOptions = 15;
```
btw thanks for going the extra mile to make this easily configurable.

File: src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace Aspire.Dashboard.Components.Controls;
+
+public partial class ResourceSelect
+{
+    private const int ResourceOptionPixelHeight = 32;
+    private const int MaxVisibleResourceOptions = 15;
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs:15]:
```suggestion
    private const int MaxVisibleResourceOptions = 15;
    private const int SelectPadding = 4; // 2px top + 2px bottom
```

File: src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            return null;
+        }
+
+        if (count <= MaxVisibleResourceOptions)
+        {
+            return null;
+        }
+
+        return $"{ResourceOptionPixelHeight * MaxVisibleResourceOptions}px";
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs:60]:
```suggestion
        return $"{(ResourceOptionPixelHeight * MaxVisibleResourceOptions) + SelectPadding}px";
```

File: src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+using Microsoft.FluentUI.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace Aspire.Dashboard.Components.Controls;
+
+public partial class ResourceSelect
+{
+    private const int ResourceOptionPixelHeight = 32;
+    private const int MaxVisibleResourceOptions = 15;
+    private const int SelectPadding = 4; // 2px top + 2px bottom
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/ResourceSelect.razor.cs:16]:
```suggestion
    private const int SelectPadding = 8; // 4px top + 4px bottom
```

**Comment by JamesNK** [comment]
I added a little more to the height to take into account padding. Will merge. Thanks!

**Pull request merged**